### ISSUE TITLE
Add shortlist of strings for support in RetinaNet compile() method.

### ISSUE THREAD

**Issue by LukeWood** [keras-team/keras-cv#807]
basically, `box_loss="l1"` `classification_loss="focal"` etc.

**Comment by martin-gorner** [comment]
and also
loss = "default"
OR
box_loss="default" classification_loss="default" 
as decided in [sync meeting](https://groups.google.com/a/google.com/g/keras-cv/c/uO8zqo14xVI)
